MC/ExtractEmbedded.C

```diff
@@ -265,6 +257,98 @@ void FilterEvent(AliESDEvent *evIn, AliESDfriend* evInF, AliESDEvent *evOut, Ali
     trcOut->SetKinkIndexes(kinkID); // reset indices
     ntrAdd++;
   }
+
+  // >>> Calo >>>
+  // filter calo clusters
+  Int_t nclIn = evIn->GetNumberOfCaloClusters(), nclOut = 0;
+  int caloClusStatus[nclIn];
+  memset(caloClusStatus,-1,sizeof(int)*nclIn);
+  for (Int_t ic=0; ic<nclIn; ic++) {
+    AliESDCaloCluster *cluster = evIn->GetCaloCluster(ic);
+    int nlab = cluster->GetNLabels();
+    TArrayI* labels = cluster->GetLabelsArray();
+    for (int il=0;il<nlab;il++) {
+      if (KeepTrack( (*labels)[il]) ) {
+        caloClusStatus[ic] = -10; // keep
+        break; // keep clusters with at least single signal contribution
```

Comment:
Is the break needed? It could be that the loop is not ended and there are still MC labels from the background not changed to -1?

Reply:
@gconesab indeed, the break was there just to avoid unnecessary checks once it is clear that the cluster should be kept, but it may leave some bg. labels unchanged. Fixed.